src/pip/__pip-runner__.py

```diff
@@ -0,0 +1,36 @@
+"""Execute exactly this copy of pip, within a different environment.
+
+This file is named as it is, to ensure that this module can't be imported via
+an import statement.
+"""
+
+import runpy
+import sys
+import types
+from importlib.machinery import ModuleSpec, PathFinder
+from os.path import dirname
+from typing import Optional, Sequence, Union
+
+PIP_SOURCES_ROOT = dirname(dirname(__file__))
+
+
+class PipImportRedirectingFinder:
+    @classmethod
+    def find_spec(
+        self,
+        fullname: str,
+        path: Optional[Sequence[Union[bytes, str]]] = None,
+        target: Optional[types.ModuleType] = None,
+    ) -> Optional[ModuleSpec]:
+        if fullname != "pip":
```

Comment:
```suggestion
        if fullname != "pip" or path is not None:
```

To be sure we are only considering the top level `pip` package here ?

Reply:
Hm, nevermind, since this is *fullname* we should be good.